### DISCUSSION TITLE
Question: How are tabs in separate panels in a single Firefox window handled when SideBery is uninstalled/disabled?

### DISCUSSION THREAD

**Discussion by davidleejy** [mbnuqw/sidebery#939] [General]
Hello fellow SideBery users, 

Opening a discussion about concerns that users may have about how tabs in separate panels will be handled after SideBery is uninstalled, disabled, or if tabs are synced to a Firefox installation without SideBery; Firefox _likely_ does not recognize the concept of "panels", so what happens to tabs stored in separate panels?

I.e. upon SideBery uninstallation/disabling, are tabs in panels beyond the first panel: (i) discarded?; (ii) rendered in a separate Firefox window?; (iii) stacked in sequential order in the same Firefox window?; etc.

Depending on the outcome, some users might prefer using native Firefox windows to partition their stacks of tabs instead of Sidebery panels.

**Comment by mbnuqw** [comment]
On sidebery uninstalling tabs will stay in sequential order in the same Firefox window. It's planned to add more options to convert sidebery panels to windows (#884) (currently in the beta version, you can only drag and drop the panel's icon to the outside of the window to create new window with tabs of the dragged panel).

**Comment by davidleejy** [comment]
Does anyone feel that tabs in separate SideBery panels should _default_ to being placed in separate Firefox windows upon SideBery uninstallation?

Having used firefox for quite some time, I feel I'm alright either way (tabs inserted into the same window or tabs inserted into separate windows) since groups of tabs can be shifted using native firefox controls to act on either result to achieve the other (e.g., A set of tabs in a single Firefox window can be selected and shifted to a separate window).

**Comment by JohnVKRDam** [comment]
I would definitely **_not_** like it if the **_default_** was to create new windows for all former Sidebery panels!

If it is an _**option**_ that can be set in Sidebery, I'm fine with it. As long as **_the default setting_** upon installing Sidebery **_is to keep them in a single window_**,

Any other methods should be an option that has to be set knowingly.

Just my €0.02  :-)

**Reply by davidleejy** [reply]
> I would definitely **_not_** like it if the **_default_** was to create new windows for all former Sidebery panels!
> 
> If it is an _**option**_ that can be set in Sidebery, I'm fine with it. As long as **_the default setting_** upon installing Sidebery **_is to keep them in a single window_**,
> 
> Any other methods should be an option that has to be set knowingly.
> 
> Just my €0.02 :-)

Does anyone feel that it might be a good idea to delineate the tab groups in a single window? For example, if tabs 1, 2, 3 were in a different Sidebery panel than tabs 4, 5, 6, 7, then after uninstalling Sidebery, perhaps a series of three empty new tabs could be used to delineate these two tab groups? So that the tabs in a single window could be like:

```
tab 1, tab 2, tab 3, new tab, new tab, new tab, tab 4, tab 5, tab 6, tab 7.
```

**Reply by JohnVKRDam** [reply]
That's kind of the method I use now in certain cases.

In between tabs that I want "grouped" for one reason or another, but not in a separate panel or folder, I stick a tab with the name:

====================================================

It is an obvious separator, which allows me to quickly identify groups of tabs.
I currently use the Firefox extension "Tab ReTitle" to change the tab name.

In the (possibly future) Sidebery uninstall options there could maybe even be a checkbox to allow the original panel name to be added to that specific separator tab name. Like this:

PANEL NAME ===========================================

So, yeah, I guess adding a tab, with a specific name, as a separator would work?

However, it could get a little complicated, since there are also actual separators and folders in Sidebery. How would you export those to the new window after uninstall? They would need to act as separators as well, I think, with the containing tabs of the folder, added below them? Should all the different depth levels (Panel > Folder > Separator) get their own separator type/name, and act as a "group" header?

What would the end result look like after the uninstall in the new window? How hard would it be to code such an uninstall option, without making the final result of tabs look chaotic? Because (only) speaking for myself, I generally have about 600 tabs open in firefox. Not all loaded/activated, of course/ Most are unloaded/dormant. But, with many tabs open, could that become problematic? I think I may be an exception(?), but in the end you also need to code for exceptions  ;-)

Anyway, these are just some random thoughts that popped into my head, while writing my reply  :-)  haha
